### ISSUE TITLE
Moes BHT-002 Thermostat not showing running_state

### ISSUE THREAD

**Issue by KameDomotics** [Koenkk/zigbee2mqtt#8698]
## What happened
In HomeAssistant thermostat and Zigbee2mqtt is not showing me the running_state

## What did you expect to happen
running_state: The current running state. Possible values are: idle, heat, cool.

## How to reproduce it (minimal and precise)
Send a message to zigbee2mqtt/FRIENDLY_NAME/get 
with payload {"running_state": ""}
## Debug info
Zigbee2MQTT version: 1.21.1-1
Adapter hardware: CC2538
Adapter firmware version:

**Comment by KameDomotics** [comment]
![A16178C1-9C87-46C8-9626-67A86B08C5B1](https://user-images.githubusercontent.com/47695661/132751312-725506f8-5669-462e-af88-5c8b853be038.jpeg)
![FA6EFBD0-F182-45BC-BC28-23770F20C090](https://user-images.githubusercontent.com/47695661/132751326-525a543a-9252-4f5b-af28-28a744795416.jpeg)
![9083EEB0-F4D3-4436-9682-9A6E4316B88E](https://user-images.githubusercontent.com/47695661/132751328-5248b797-a88f-4c2f-8a25-a2d5bc0f4a65.png)

**Comment by Plawasan** [comment]
That would probably explain this error that keeps showing up in my logs:

Logger: homeassistant.helpers.template
Source: helpers/template.py:1485
First occurred: 8:47:01 (24 occurrences)
Last logged: 9:06:17

Template variable warning: dict object has no element Undefined when rendering '{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}'

**Comment by KameDomotics** [comment]
Is there any way to fix this?

**Comment by Safs** [comment]
Same problem for me too...

**Comment by rizer1980** [comment]
same problem. log file grow up hundred megabytes...
5 BHT-002-GCLZB thermostat
with debounce:
Logger: homeassistant.helpers.template
Source: helpers/template.py:1485
First occurred: 1:26:56 (3682 occurrences)
Last logged: 10:41:51

Template variable warning: dict object has no element Undefined when rendering '{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}'

**Comment by KameDomotics** [comment]
> same problem. log file grow up hundred megabytes...
> 5 BHT-002-GCLZB thermostat
> with debounce:
> Logger: homeassistant.helpers.template
> Source: helpers/template.py:1485
> First occurred: 1:26:56 (3682 occurrences)
> Last logged: 10:41:51
> 
> Template variable warning: dict object has no element Undefined when rendering '{% set values = {'idle':'off','heat':'heating','cool':'cooling','fan only':'fan'} %}{{ values[value_json.running_state] }}'

that's right, it gives me the same warning

**Comment by kocsishome** [comment]
That's right.  Is there a solution to this?  Thanks

**Comment by jinnerbichler** [comment]
Same for me.

The state send over MQTT has a field with the name `head`, which has the value 'ON' or 'OFF'.
So the data for `running_state` in available, but is not sent via `running_state`

```
{
  "child_lock": "UNLOCK",
  "current_heating_setpoint": 12,
  "deadzone_temperature": 1,
  "heat": "OFF",
  "linkquality": 135,
  "local_temperature": 26,
  "local_temperature_calibration": -3,
  "max_temperature": 35,
  "max_temperature_limit": 45,
  "preset": "hold",
  "preset_mode": "hold",
  "program": [
    {
      "weekdays_p1": "6h:0m 20°C"
    },
    {
      "weekdays_p2": "11h:30m 21°C"
    },
    {
      "weekdays_p3": "13h:30m 22°C"
    },
    {
      "weekdays_p4": "17h:30m 23°C"
    },
    {
      "saturday_p1": "6h:0m 24°C"
    },
    {
      "saturday_p2": "12h:0m 23°C"
    },
    {
      "saturday_p3": "14h:30m 22°C"
    },
    {
      "saturday_p4": "17h:30m 21°C"
    },
    {
      "sunday_p1": "6h:0m 19°C"
    },
    {
      "sunday_p2": "12h:30m 20°C"
    },
    {
      "sunday_p3": "14h:30m 21°C"
    },
    {
      "sunday_p4": "18h:30m 20°C"
    }
  ],
  "sensor": "AL",
  "system_mode": "heat"
}
```

**Comment by KameDomotics** [comment]
> Same for me.
> 
> The state send over MQTT has a field with the name `head`, which has the value 'ON' or 'OFF'.
> So the data for `running_state` in available, but is not sent via `running_state`
> 
> ```
> {
>   "child_lock": "UNLOCK",
>   "current_heating_setpoint": 12,
>   "deadzone_temperature": 1,
>   "heat": "OFF",
>   "linkquality": 135,
>   "local_temperature": 26,
>   "local_temperature_calibration": -3,
>   "max_temperature": 35,
>   "max_temperature_limit": 45,
>   "preset": "hold",
>   "preset_mode": "hold",
>   "program": [
>     {
>       "weekdays_p1": "6h:0m 20°C"
>     },
>     {
>       "weekdays_p2": "11h:30m 21°C"
>     },
>     {
>       "weekdays_p3": "13h:30m 22°C"
>     },
>     {
>       "weekdays_p4": "17h:30m 23°C"
>     },
>     {
>       "saturday_p1": "6h:0m 24°C"
>     },
>     {
>       "saturday_p2": "12h:0m 23°C"
>     },
>     {
>       "saturday_p3": "14h:30m 22°C"
>     },
>     {
>       "saturday_p4": "17h:30m 21°C"
>     },
>     {
>       "sunday_p1": "6h:0m 19°C"
>     },
>     {
>       "sunday_p2": "12h:30m 20°C"
>     },
>     {
>       "sunday_p3": "14h:30m 21°C"
>     },
>     {
>       "sunday_p4": "18h:30m 20°C"
>     }
>   ],
>   "sensor": "AL",
>   "system_mode": "heat"
> }
> ```

In "developer tools" also appears the status `running_state: idle / heat`, so I don't understand why it is not displayed in the card and generates the warning

**Comment by KameDomotics** [comment]
> Same for me.
> 
> The state send over MQTT has a field with the name `head`, which has the value 'ON' or 'OFF'.
> So the data for `running_state` in available, but is not sent via `running_state`
> 
> ```
> {
>   "child_lock": "UNLOCK",
>   "current_heating_setpoint": 12,
>   "deadzone_temperature": 1,
>   "heat": "OFF",
>   "linkquality": 135,
>   "local_temperature": 26,
>   "local_temperature_calibration": -3,
>   "max_temperature": 35,
>   "max_temperature_limit": 45,
>   "preset": "hold",
>   "preset_mode": "hold",
>   "program": [
>     {
>       "weekdays_p1": "6h:0m 20°C"
>     },
>     {
>       "weekdays_p2": "11h:30m 21°C"
>     },
>     {
>       "weekdays_p3": "13h:30m 22°C"
>     },
>     {
>       "weekdays_p4": "17h:30m 23°C"
>     },
>     {
>       "saturday_p1": "6h:0m 24°C"
>     },
>     {
>       "saturday_p2": "12h:0m 23°C"
>     },
>     {
>       "saturday_p3": "14h:30m 22°C"
>     },
>     {
>       "saturday_p4": "17h:30m 21°C"
>     },
>     {
>       "sunday_p1": "6h:0m 19°C"
>     },
>     {
>       "sunday_p2": "12h:30m 20°C"
>     },
>     {
>       "sunday_p3": "14h:30m 21°C"
>     },
>     {
>       "sunday_p4": "18h:30m 20°C"
>     }
>   ],
>   "sensor": "AL",
>   "system_mode": "heat"
> }
> ```

In "developer tools" also appears the status `running_state: idle / heat`, so I don't understand why it is not displayed in the card and generates the warning

![Uploading 5E48DD3A-8105-4817-B09E-DE194F0CC5B8.jpeg…]()

**Comment by KameDomotics** [comment]
![7CD6907C-E4C4-4A33-92D8-FC555C44697E](https://user-images.githubusercontent.com/47695661/134788561-98f17959-583b-476e-a727-ba63c075c270.jpeg)

**Comment by rizer1980** [comment]
to disable this flood warning
in /config/configuration.yaml add this:

logger:
  default: warning
  logs:
    homeassistant.helpers.template: error

**Comment by jinnerbichler** [comment]
I think this issue is addressed in this merge: https://github.com/Koenkk/zigbee-herdsman-converters/pull/3057

**Comment by KameDomotics** [comment]
> I think this issue might be addressed in this merge: [Koenkk/zigbee-herdsman-converters#3057](https://github.com/Koenkk/zigbee-herdsman-converters/pull/3057)
> I guess we just have to wait for next release.

That would be really cool!  Thank you very much 😃

**Comment by KameDomotics** [comment]
Problem solved with Zigbee2mqtt v1.21.2-1

Thanks!

**Comment by Plawasan** [comment]
> Problem solved with Zigbee2mqtt v1.21.2-1
> 
> Thanks!

You mean the dev branch?
